core/drivers/crypto/crypto_api/acipher/dsa.c

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2018-2021 NXP
+ *
+ * Crypto DSA interface implementation to enable HW driver.
+ */
+#include <crypto/crypto.h>
+#include <drvcrypt.h>
+#include <drvcrypt_acipher.h>
+
+/*
+ * Get the recommended L and N bits parameters corresponding
+ * respectively to the size of the Primes P and G (and so
+ * the Public Key and Private Key).
+ *
+ * Refer the NIST.FIPS 186-4 section 4.2
+ *
+ * @size_bits   Maximum key size bits
+ * @l_bits      [out] L size in bits
+ * @n_bits      [out] N size in bits
+ */
+static TEE_Result get_keys_size(size_t size_bits, size_t *l_bits,
+				size_t *n_bits)
+{
+	if (size_bits <= 1024)
+		*n_bits = 160;
+	else if (size_bits <= 2048)
+		*n_bits = 256;
```

Comment:
I guess this conditional statement has been added for cases ```L=2048 N=224``` and ```L=2048 N=256```. Since we went for the pair ```L=2048 N=256```, I guess we can merge it with ```L=3072 N=256```